### PR TITLE
BIGTOP-3915:soft link error about jar of hadoop in {hbase_home}/lib

### PR DESCRIPTION
**Pull request by guluo2016**
### Description of PR

Solve soft link error ,

Details see: [BIGTOP-3915](https://issues.apache.org/jira/browse/BIGTOP-3915)

### How was this patch tested?
building hbase rpm,
installing hbase,
check soft link about hadoop  in {hbase_home}/lib

### For code changes:
rpm: hbase.spec
deb: hbase/rules

Please review @iwasakims thanks

### CODE REVIEW

**Comment by iwasakims** [comment]
> In addition, i found hbase is still normal even thoes soft link error, this is puzzling.

It works because [the classpath set by hbase CLI contains classpath of Hadoop](https://github.com/apache/hbase/blob/rel/2.4.13/bin/hbase#L337-L352), I guess.

```
[root@5593b54af887 hbase]# hbase classpath | sed -z 's/:/\n/g'
/etc/hbase/conf
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b09-2.el8_7.x86_64/lib/tools.jar
/usr/lib/hbase
/usr/lib/hbase/lib/shaded-clients/hbase-shaded-client-byo-hadoop-2.4.13.jar
/usr/lib/hbase/lib/client-facing-thirdparty/audience-annotations-0.5.0.jar
/usr/lib/hbase/lib/client-facing-thirdparty/commons-logging-1.2.jar
/usr/lib/hbase/lib/client-facing-thirdparty/htrace-core4-4.2.0-incubating.jar
/usr/lib/hbase/lib/client-facing-thirdparty/jcl-over-slf4j-1.7.33.jar
/usr/lib/hbase/lib/client-facing-thirdparty/jul-to-slf4j-1.7.33.jar
/usr/lib/hbase/lib/client-facing-thirdparty/reload4j-1.2.19.jar
/usr/lib/hbase/lib/client-facing-thirdparty/slf4j-api-1.7.33.jar
/usr/lib/hbase/lib/client-facing-thirdparty/slf4j-reload4j-1.7.33.jar
/etc/hadoop/conf
/usr/lib/hadoop/lib/*
/usr/lib/hadoop/.//*
/usr/lib/hadoop-hdfs/./
/usr/lib/hadoop-hdfs/lib/*
/usr/lib/hadoop-hdfs/.//*
/usr/lib/hadoop-mapreduce/.//*
/usr/lib/hadoop-yarn/./
/usr/lib/hadoop-yarn/lib/*
/usr/lib/hadoop-yarn/.//*
/etc/hadoop/conf
```

Let me investigate whether we still need those symlinks.

File: bigtop-packages/src/deb/hbase/rules [hunk 1]
```diff
@@ -50,15 +53,15 @@ override_dh_auto_install:
...
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-core.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-jobclient.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
-	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-shuffle.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
+	ln -f -s ${hadoop_mapreduce_path}/hadoop-mapreduce-client-shuffle.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-yarn-api.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-yarn-client.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
-	ln -f -s ${hadoop_home}/client/hadoop-yarn-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
+	ln -f -s ${hadoop_yarn_path}/hadoop-yarn-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-yarn-server-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
```
Comment (iwasakims) [inline bigtop-packages/src/deb/hbase/rules:65]: hadoop-yarn-server-common.jar needs the same fix? I tested the patch like following. @guluo2016 

```
[root@ec68ac35bd71 /]# for f in $(find /usr/lib/hbase/lib -type l) ; do test -e "$f" || echo "$f does not exist." ; done
/usr/lib/hbase/lib/hadoop-yarn-server-common.jar does not exist.
```

**Comment by iwasakims** [comment]
> Let me investigate whether we still need those symlinks.

Anyway we can commit the patch here before further refactoring. The patch looks harmless.

File: bigtop-packages/src/deb/hbase/rules [hunk 2]
```diff
@@ -50,15 +53,15 @@ override_dh_auto_install:
...
+	ln -f -s ${hadoop_mapreduce_path}/hadoop-mapreduce-client-app.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-core.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-jobclient.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
-	ln -f -s ${hadoop_home}/client/hadoop-mapreduce-client-shuffle.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
+	ln -f -s ${hadoop_mapreduce_path}/hadoop-mapreduce-client-shuffle.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-yarn-api.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
 	ln -f -s ${hadoop_home}/client/hadoop-yarn-client.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
-	ln -f -s ${hadoop_home}/client/hadoop-yarn-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
+	ln -f -s ${hadoop_yarn_path}/hadoop-yarn-common.jar debian/tmp/usr/lib/${hbase_pkg_name}/lib/
```
Comment (iwasakims) [inline bigtop-packages/src/deb/hbase/rules:64]: hadoop-yarn-common.jar seems not to need the fix.

Comment (guluo2016) [reply bigtop-packages/src/deb/hbase/rules:64]: Thank you for your patient review, you are right,I will modify it again later

**Review by iwasakims — APPROVED** [review]
LGTM. Thanks for the contribution, @guluo2016.

**Pull request merged**